File: JavaScript/12.concatArray.js
/* Leetcode 1929. Concatenation of Array*/
/*Given an integer array nums of length n, you want to create an array ans of length 2n where ans[i] == nums[i] and ans[i + n] == nums[i] for 0 <= i < n (0-indexed).
Specifically, ans is the concatenation of two nums arrays.
Return the array ans. */

//method 1
/**
 * @param {number[]} nums
 * @return {number[]}
 */
 var getConcatenation = function(nums) {
     
    return nums.concat(nums)
};

getConcatenation([1,2,1])
getConcatenation([1,3,2,1])

//method 2
var getConcatenation = function(nums) {
    var ans = []
    
    ans.push(...nums, ...nums)
    return ans
};

//method 3
var getConcatenation = function(nums) {
    var ans = []
    
    for (var i = 0; i < nums.length; i++) {
      ans.push(nums[i])
    }
  	for (var i = 0; i < nums.length; i++) {
      ans.push(nums[i])
    }
    return ans
};

File: Java/12concatArray.java
/* Leetcode 1929. Concatenation of Array */
/* Given an integer array nums of length n, you want to create an array ans of length 2n where ans[i] == nums[i] and ans[i + n] == nums[i] for 0 <= i < n (0-indexed).
Specifically, ans is the concatenation of two nums arrays.
Return the array ans. */

public class Main {
    
    public static int[] getConcatenation(int[] nums) {
		
        int[] ans = new int[nums.length * 2];
        
        for (int i = 0; i < nums.length; i ++) {
            ans[i] = nums[i];
        }
        
        for (int j = ans.length/2; j < ans.length; j++) {
            ans[j] = nums[j-ans.length/2];
        }
        
        return ans;
        }
    
    public static void main(String[] args) {
        
        int[] arr1 = {1,2,1};
        
        getConcatenation(arr1);
    
    }
}
